nistats/second_level_model.py

```diff
@@ -0,0 +1,451 @@
+    """ Implementation of the General Linear Model for multiple subject
+    fMRI data
+
+    Parameters
+    ----------
+
+    mask: Niimg-like, NiftiMasker or MultiNiftiMasker object, optional,
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask will be used. If no mask is given,
+        it will be computed automatically by a MultiNiftiMasker with default
+        parameters. Mask automatic computation assumes first level imgs have
+        already been masked.
+
+    smoothing_fwhm: float, optional
+        If smoothing_fwhm is not None, it gives the size in millimeters of the
+        spatial smoothing to apply to the signal.
+
+    memory: string, optional
+        Path to the directory used to cache the masking process and the glm
+        fit. By default, no caching is done. Creates instance of joblib.Memory.
+
+    memory_level: integer, optional
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    verbose : integer, optional
+        Indicate the level of verbosity. By default, nothing is printed.
+
+    n_jobs : integer, optional
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs', -2 'all CPUs but one', and so on.
+
+    """
+    def __init__(self, mask=None, smoothing_fwhm=None,
+                 memory=None, memory_level=1, verbose=1,
+                 n_jobs=1, minimize_memory=True):
+        self.mask = mask
+        self.smoothing_fwhm = smoothing_fwhm
+        self.memory = memory
+        self.memory_level = memory_level
+        self.verbose = verbose
+        self.n_jobs = n_jobs
+        self.minimize_memory = minimize_memory
+        self.labels_ = None
+        self.results_ = None
+
+    def fit(self, first_level_input, first_level_conditions=None,
+            confounds=None, design_matrix=None,
+            first_level_conditions_name=None):
+        """ Fit the GLM
+
+        1. create design matrix
+        2. do a masker job: fMRI_data -> Y
+        3. fit regression to (Y, X)
+
+        Parameters
+        ----------
+        first_level_input: list of `FirstLevelModel` objects or pandas
+                           DataFrame or list of Niimg-like objects.
+            If list of `FirstLevelModel` objects, then first_level_conditions
+            must be provided. If a pandas DataFrame, then must contain columns
+            subject_id, map_name, effects_map_path. If list of Niimg-like
+            objects then this will be taken literally as Y for the model fit
+            and design_matrix must be provided.
+
+        first_level_conditions: list of (str or array) or None
+            If first_level_input is a list of `FirstLevelModel` objects then
+            it is mandatory to provide a list of contrast definitions to pass
+            to the compute_contrast method of `FirstLevelModel` which can be
```

Comment:
avoid 'which can be', because it is not obvious what 'which' refers to in such nested sentences.